src/App.tsx

```diff
@@ -1,54 +1,135 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import './App.scss';
+import classNames from 'classnames';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [isInputFocused, setIsInputFocused] = useState(false);
+
```

Comment:
![image](https://github.com/mate-academy/react_autocomplete/assets/91946036/0314167d-ce94-43ed-a557-640636480924)

unfortunately this part was not implemented

when you type something it should be a delay before you show the results in dropdown